inveniordm_py/files/metadata.py

```diff
@@ -13,24 +13,69 @@
 class FileMetadata(Metadata):
     """File metadata."""
 
-    accept = "application/vnd.inveniordm.v1+json"
-    content_type = "application/vnd.inveniordm.v1+json"
+    accept = "application/json"
+    content_type = "application/json"
 
     @property
     def endpoint_kwargs(self):
         """Return endpoint kwargs for a file, namely the filename."""
-        return {"filename": self._data["filename"]}
+        return {"key": self._data["key"]}
 
 
 class FilesListMetadata(ListMetadata):
     """Metadata of a list of files."""
 
-    accept = "application/vnd.inveniordm.v1+json"
+    accept = "application/json"
+    content_type = "application/json"
 
     item_class = FileMetadata
 
+    def __init__(self, data=None, **kwargs):
+        """Initialize metadata object.
 
-class Stream:
+        The data is either explicitely passed (i.e. top-level parameters) or taken from keyword arguments.
+
+        .. note:: this is a bit of a hack, but it works for now.
+        """
+        if data:
```

Comment:
This is hackish, the issue was that the endpoint to initialize expects a top-level parameter like so:

```shell
POST /api/records/{id}/draft/files HTTP/1.1
Content-Type: application/json

[
  {"key": "figure.png"},
  {"key": "article.pdf"},
  {"key": "data.zip"}
]

```

while other resources and endpoints accept named parameters, e.g. to upload a draft file it expects the parameter `filename` in the body.